infra/modules/azure_function_app/variables.tf

```diff
@@ -0,0 +1,86 @@
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "env_short" {
+  type        = string
+  description = "Environment short name"
+}
+
+variable "location_short" {
+  type        = string
+  description = "Location short name"
+}
+
+variable "location" {
+  type        = string
+  description = "Location name"
+}
+
+variable "prefix" {
+  type        = string
+  description = "Project prefix"
+}
+
+variable "instance_number" {
+  type        = string
+  description = "(Optional) Instance count for a specific resource"
+  default     = "01"
+}
+
+variable "domain" {
+  type        = string
+  description = "Domain of the project"
+}
+
+variable "app_name" {
+  type        = string
+  description = "Application name"
+}
+
+variable "resource_group_name" {
+  type = string
+}
+
+variable "application_insights_connection_string" {
+  type      = string
+  sensitive = true
+}
+
+variable "health_check_path" {
+  type = string
+}
+
+variable "node_version" {
+  type = number
```

Comment:
can we force an enum? moreover, I'm wondering if this module should be called 

`azure_function_app_node` since a function supports generic runtimes (ie. java). for example self-care uses java functions. if the moving parts (variables to setup rintime) aren't that much, we may consider to collapse both into one module.

Reply:
yes we can, although enums in HCL are quite awkward

it's a question I asked to myself as well. We should add something like [this](https://github.com/pagopa/terraform-azurerm-v3/blob/c5e3aade01153d9dbd0bfb4f32bc6834fbded3c2/function_app/main.tf#L257), including node, java and docker only

Reply:
I would remove this variable, to me we should support only one node version: the latest LTS available (currently 20.x)

Reply:
@lucacavallaro I'm afraid this approach would be a little too tight. What if you need to make a change and suddenly the module support the new version only? We should use tags to version the module, but I'd avoid that approach as long as we can (see Slack)

Reply:
> it's a question I asked myself as well. We should add something like this, including node, java and docker only

If i remember well there were some issues using docker on app services @lucacavallaro

Reply:
As discussed, Docker had issue with lifecycle and log management. We are going to include node and java only